src/Functions/FunctionsConsistentHashing.h

```diff
@@ -43,7 +44,7 @@ class FunctionConsistentHashImpl : public IFunction
 
     DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
     {
-        if (!isInteger(arguments[0]))
+        if (!isInteger(arguments[0]) && !isIPv4(arguments[0]))
```

Comment:
+ ` && !isIPv6(arguments[0])`?

Reply:
I made IPv6 and IPv4 to be represented by Numper, Integer and UnsignedInteger - see DataTypeIPv4andIPv6.h - for IPv4 it's definitely true, but not sure about IPv6...